### PR TITLE
[wip] [Design] XGBoost on Katib

### PR DESCRIPTION
**Pull request by samplise**
Create a design doc for katib xgboost under /doc/design. It includes initial design and will be updated as project progress.

### CODE REVIEW

File: doc/design/design_katib_xgboost.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# SQLFlow Trains Models via XGBoost in Katib
+
+## Requirement description:
+
+SQLFlow needs to provide users with a simple way to train their ML model. This function is supported by SQLFlow running XGBoost jobs on Katib.
```
Comment (ywskycn) [inline doc/design/design_katib_xgboost.md]: - What SQLFlow provides?
- How an existing SQLFlow sql looks like? Users need to specify model, hyper-parameters, input data, etc.
- This doc is to further reduce users' inputs by introducing hpo, to remove `with`.

File: doc/design/design_katib_xgboost.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## Hyperparameter Optimization in Katib
+
+[Katib](https://github.com/kubeflow/katib) is a Kubernetes Native System for Hyperparameter Tuning and Neural Architecture Search. The system is inspired by Google vizier and supports multiple ML/DL frameworks (e.g. TensorFlow, MXNet, and PyTorch).
```
Comment (terrytangyuan) [inline doc/design/design_katib_xgboost.md]: Apache MXNet

File: doc/design/design_katib_xgboost.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+## Support XGBoost job in Katib
+
+[XGBoost](https://xgboost.readthedocs.io/en/latest/) is an optimized distributed gradient boosting library designed to be highly efficient, flexible and portable. 
+
+When to run XGBoost jobs in Katib, we create an Experiment CR on Kubernetes via a yaml file. This Experiment CR will create one Suggestion CR and multiple Trail Pods later. The Suggestion CR generates the value for hyperparameters. Each Trail Pod includes two containers: a job container and a MetricsCollector container. The job container is created from a standard XGBoost Docker image created by us. When the job container is started, it receives parameters defined in SQLFlow as well as the value of hyperparameters from Suggestion CR.
```
Comment (terrytangyuan) [inline doc/design/design_katib_xgboost.md]: Trail -> Trial

File: doc/design/design_katib_xgboost.md (hunk 2 above)
Comment (terrytangyuan) [inline doc/design/design_katib_xgboost.md]: vizier -> Vizier

**Pull request closed without merging**